### PR TITLE
Sprint Challenge: React - Star Wars

### PR DESCRIPTION
**Pull request by tmshkr**

### CODE REVIEW

**Comment by tmshkr** [comment]
[Live Site](https://tmshkr.github.io/Sprint-Challenge-React-Wars/)

File: Answers.md [hunk 1]
```diff
@@ -1,11 +1,38 @@
...
+  
+## Describe props.
+In React, data is passed down from parent to child components through named attributes called props. Each child component has a props object passed down to it, which has references to the data passed from the parent component. Props are read-only, so that a child component should not modify the prop values that are passed to it.
+
+## What are side effects, and how do you sync effects in a React component to state or prop changes?
+In computer science, a side effect is said to occur when an operation has an observable effect other than its main effect, which is the value it returns.  In React, `useEffect` produces a side effect for a functional component, allowing the developer to write code that will execute after the initial render. In order to sync side effects with changes to state or props, a dependency array is passed to the `useEffect` hook, which lets the hook know to fire after the component re-renders in response to changes in the dependencies passed to the array.
+
```
Comment (jskway) [inline Answers.md:34]: 💯

File: starwars/src/App.js [hunk 2]
```diff
@@ -1,19 +1,51 @@
...
+          <>
+            <div></div>
+            <Spinner
+              type="grow"
+              style={{
+                color: "#fff",
+                width: "3rem",
+                height: "3rem",
+                margin: "3rem auto"
+              }}
```
Comment (jskway) [inline starwars/src/App.js:41]: Nice use of a Spinner :+1:

File: starwars/src/App.js [hunk 3]
```diff
@@ -1,19 +1,51 @@
...
+  // Fetch characters from the star wars api in an effect hook. Remember, anytime you have a
   // side effect in a component, you want to think about which state and/or props it should
   // sync up with, if any.
+  useEffect(() => {
+    const url = `https://swapi.co/api/people/?page=${currentPage}&format=json`;
+    axios.get(url).then(({ data }) => {
+      console.log(data);
+      setData(data);
+    });
+  }, [currentPage]);
```
Comment (jskway) [inline starwars/src/App.js:23]: Great job with the dynamic url and request

File: starwars/src/components/Paginator.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        <PaginationLink
+          next
+          onClick={() => handlePage(currentPage + 1)}
+          disabled={currentPage === totalPages}
+        />
+      </PaginationItem>
+    </Pagination>
+  );
+}
+
```
Comment (jskway) [inline starwars/src/components/Paginator.js:58]: Nice work on this stretch goal -- impressive :+1:

**Review by jskway — APPROVED** [review]

**Pull request merged**